### DISCUSSION TITLE
Model not training - Acts likes it training, saves 8.6gb checkpoints, but they are the same as the base model

### DISCUSSION THREAD

**Discussion by minienglish1** [d8ahazard/sd_dreambooth_extension#1062] [Q&A] [answered]
WebUI-user start up output and config.json file at end.

Everything looks like it is working.  Training images get bucketed and cached, training starts and finished, checkpoints are saved.  However, the output checkpoints are 8.6gb, instead of the usual ~4gb.  Model sample images at the exact same as the base model.  So the model is not being trained.

Cleared out the venv folder an let it rebuild itself.  Not sure what is going on.

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Dreambooth revision: bff61c6b92b79ece2e140ee240e0628dd9a4ebef
SD-WebUI revision: 3c922d983bf60ba187b5422b3690e6b7fb07777e

Checking Dreambooth requirements...
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64"' don't match your environment
Ignoring mediapipe-silicon: markers 'sys_platform == "darwin"' don't match your environment
Collecting accelerate==0.16.0
  Using cached accelerate-0.16.0-py3-none-any.whl (199 kB)
Collecting gitpython~=3.1.31
  Using cached GitPython-3.1.31-py3-none-any.whl (184 kB)
Collecting transformers~=4.26.1
  Using cached transformers-4.26.1-py3-none-any.whl (6.3 MB)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Installing collected packages: requests, gitpython, accelerate, transformers
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Uninstalling requests-2.25.1:
      Successfully uninstalled requests-2.25.1
  Attempting uninstall: gitpython
    Found existing installation: GitPython 3.1.30
    Uninstalling GitPython-3.1.30:
      Successfully uninstalled GitPython-3.1.30
  Attempting uninstall: accelerate
    Found existing installation: accelerate 0.12.0
    Uninstalling accelerate-0.12.0:
      Successfully uninstalled accelerate-0.12.0
  Attempting uninstall: transformers
    Found existing installation: transformers 4.25.1
    Uninstalling transformers-4.25.1:
      Successfully uninstalled transformers-4.25.1
Successfully installed accelerate-0.16.0 gitpython-3.1.31 requests-2.28.2 transformers-4.26.1

[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
[+] accelerate version 0.16.0 installed.
[+] bitsandbytes version 0.35.4 installed.
[+] diffusers version 0.13.1 installed.
[+] transformers version 4.26.1 installed.
[+] xformers version 0.0.17.dev464 installed.

#######################################################################################################

"adamw_weight_decay": 0.01,
    "attention": "xformers",
    "cache_latents": true,
    "clip_skip": 1,
    "concepts_list": [
        {
            "class_data_dir": "C:\\Users\\3090\\stable-diffusion-webui\\models\\dreambooth\\Test782-1e-6-1024-0.5text-shuffle\\classifiers_0",
            "class_guidance_scale": 1,
            "class_infer_steps": 10,
            "class_negative_prompt": "",
            "class_prompt": "",
            "class_token": "",
            "instance_data_dir": "c:\\test782",
            "instance_prompt": "[filewords]",
            "instance_token": "",
            "is_valid": true,
            "n_save_sample": 0,
            "num_class_images_per": 0,
            "sample_seed": -1,
            "save_guidance_scale": 1,
            "save_infer_steps": 10,
            "save_sample_negative_prompt": "",
            "save_sample_prompt": "",
            "save_sample_template": ""
        }
    ],
    "concepts_path": "",
    "custom_model_name": "",
    "deis_train_scheduler": false,
    "deterministic": false,
    "ema_predict": false,
    "epoch": 1,
    "epoch_pause_frequency": 0,
    "epoch_pause_time": 0,
    "freeze_clip_normalization": false,
    "gradient_accumulation_steps": 1,
    "gradient_checkpointing": false,
    "gradient_set_to_none": true,
    "graph_smoothing": 50,
    "half_model": false,
    "train_unfrozen": true,
    "has_ema": false,
    "hflip": false,
    "infer_ema": false,
    "initial_revision": 0,
    "learning_rate": 1e-06,
    "learning_rate_min": 1e-06,
    "lifetime_revision": 0,
    "lora_learning_rate": 0.0001,
    "lora_model_name": "",
    "lora_unet_rank": 4,
    "lora_txt_rank": 4,
    "lora_txt_learning_rate": 5e-05,
    "lora_txt_weight": 1,
    "lora_weight": 1,
    "lr_cycles": 1,
    "lr_factor": 0.5,
    "lr_power": 1,
    "lr_scale_pos": 0.5,
    "lr_scheduler": "constant_with_warmup",
    "lr_warmup_steps": 0,
    "max_token_length": 150,
    "mixed_precision": "fp16",
    "model_name": "Test782-1e-6-1024-0.5text-shuffle",
    "model_dir": "C:\\Users\\3090\\stable-diffusion-webui\\models\\dreambooth\\Test782-1e-6-1024-0.5text-shuffle",
    "model_path": "C:\\Users\\3090\\stable-diffusion-webui\\models\\dreambooth\\Test782-1e-6-1024-0.5text-shuffle",
    "num_train_epochs": 100,
    "offset_noise": 0,
    "optimizer": "8Bit Adam",
    "pad_tokens": true,
    "pretrained_model_name_or_path": "C:\\Users\\3090\\stable-diffusion-webui\\models\\dreambooth\\Test782-1e-6-1024-0.5text-shuffle\\working",
    "pretrained_vae_name_or_path": null,
    "prior_loss_scale": false,
    "prior_loss_target": 100.0,
    "prior_loss_weight": 0.75,
    "prior_loss_weight_min": 0.1,
    "resolution": 1024,
    "revision": 782,
    "sample_batch_size": 1,
    "sanity_prompt": "",
    "sanity_seed": 420420.0,
    "save_ckpt_after": true,
    "save_ckpt_cancel": false,
    "save_ckpt_during": true,
    "save_ema": true,
    "save_embedding_every": 1,
    "save_lora_after": true,
    "save_lora_cancel": false,
    "save_lora_during": true,
    "save_lora_for_extra_net": true,
    "save_preview_every": 0,
    "save_safetensors": true,
    "save_state_after": true,
    "save_state_cancel": false,
    "save_state_during": true,
    "scheduler": "DDIM",
    "shuffle_tags": true,
    "snapshot": "",
    "split_loss": true,
    "src": "C:\\Users\\3090\\stable-diffusion-webui\\models\\Stable-diffusion\\v2-1_768-nonema-pruned.safetensors",
    "stop_text_encoder": 0.5,
    "strict_tokens": true,
    "tf32_enable": false,
    "train_batch_size": 1,
    "train_imagic": false,
    "train_unet": true,
    "use_concepts": false,
    "use_ema": true,
    "use_lora": false,
    "use_lora_extended": false,
    "use_subdir": true,
    "v2": true
}

**Comment by minienglish1** [comment, marked as answer]
I think I figured out the problem.  Turning off "save diffusers snapshots" fixed the problem.  How should I test this issue before submitting an issue?

**Reply by Arilziem** [reply]
Huh. Just to be sure, are you using the automatic checkpoint generation or the manual one? With the manual one, I think you need to select the checkpoint before pressing the button.

Assuming you are aware of that, I would check system RAM usage, especially during snapshot and checkpoint generation. There have been a few RAM-related problems lately where stuff wasn't properly saved.

**Reply by Arilziem** [reply]
~Ok this is ... interesting. The OOM issues I [mentioned](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/1048#discussioncomment-5292752) in the other thread are gone when I disable all snapshot options. I guess it needs more VRAM as well?~ Oh never mind, I accidently disabled training unet, OOM still alive and kicking. -.-

**Comment by minienglish1** [comment]
My vram should be fine, I have 24gb.  Rarely use above 20 via task manager.  But I'll check again.

I was using the automatic snapshot generation.  Just finished training for 100 epochs without automatic snapshots, everything worked fine.  I'll test again with snapshots later to confirm.

**Reply by Arilziem** [reply]
I mean system RAM actually, _RAM_ RAM ;) I would have to look it up, but I'm pretty sure the snapshot compilation is not done on GPU.

**Reply by minienglish1** [reply]
32GB RAM + 16GB page.   Committed is high at 35.5GB, but actual in usage is only 14GB. Same during training as as when saving checkpoints.  I'll test saving snapshots later.  Still running an ancient DDR3 system, so can't add any more ram.  So if total ram is the problem, I'm SOL until I build a new system.

How much ram do you need for training anyway?  I've heard 1gb RAM per 1gb VRAM.  Don't know how it scales with multiple GPUs though.

**Reply by minienglish1** [reply]
32GB RAM + 16GB page.   Committed is high at 35.5GB, but actual in usage is only 14GB. Same during training as as when saving checkpoints.  I'll test saving snapshots later.  Still running an ancient DDR3 system, so can't add any more ram.  So if total ram is the problem, I'm SOL until I build a new system.

How much ram do you need for training anyway?  I've heard 1gb RAM per 1gb VRAM.  Don't know how it scales with multiple GPUs though.

**Reply by Arilziem** [reply]
I don't know the ratio, but I'm reminded of #1000 where 32GB effective RAM (WSL2 on a 64GB machine) was apparently low enough to cause problems _sometimes_.

**Reply by minienglish1** [reply]
That creates a serious issue for me then.  I need to figure out this ram ratio issue soon.  

Just did some quick research on ram requirements.  Lowest recommendation is equal to highest GPU vram.  Highest recommendation is 2x total GPU memory. 

I currently planning my school's new AI workstation, which must be 100% stable.  I planned for single xeon cpu, 128GB system ram, with either 3 @ RTX A6000 (48gb) or 2 @ a100 (80gb).  If 32gb ram isn't enough for a single 24g vram card, 128gb ram isn't going to be enough for the workstation.   I'll have to switch to 256gb, or maybe higher if I can find a workstation motherboard that will support it.
